src/cron.ts

```diff
@@ -98,24 +103,40 @@ export const getAllRFCRemarks = async (
           );
           continue;
         }
-      } else if (refInfo.approved) {
-        logger.debug(`Referendum has been approved: ${JSON.stringify(refInfo)}`);
-        const [approvalDate] = refInfo.approved;
-        const blockDate = await getBlockDate(approvalDate, api);
+      } else {
+        const extractDate = (): number | null => {
+          const { approved, rejected, timedOut } = refInfo;
+          if (approved) {
+            return approved[0];
+          } else if (rejected) {
+            return rejected[0];
+          } else if (timedOut) {
+            return timedOut[0];
+          }
```

Comment:
```ts
const { approved, rejected, timedOut } = refInfo;
const date = approved?.[0] || rejected?.[0] || timedOut?.[0] || null;
``` 
?

Reply:
I like it! Tho I think it has to be with `??`: `const date = approved?.[0] ?? rejected?.[0] ?? timedOut?.[0] ?? null;`